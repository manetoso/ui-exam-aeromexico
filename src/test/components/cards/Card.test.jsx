const { configureStore } = require('@reduxjs/toolkit');
const { render, screen } = require('@testing-library/react');

const { Provider } = require('react-redux');
const { favlistReducer } = require('../../../redux/slices/favlistSlice');

const { Card } = require('../../../components/cards/Card');

/**
 * TESTS FOR THE Card.jsx COMPONENT
 */
describe('Testing in Card.jsx', () => {
  // FAKE DATA FOR THE CARD
  const fakeData = {
    id: 8,
    name: 'Harry Potter',
    gender: 'male',
    house: 'Gryffindor',
    dateOfBirth: '31-07-1980',
    eyeColour: 'green',
    hairColour: 'black',
    hogwartsStudent: true,
    alive: true,
    image: 'http://hp-api.herokuapp.com/images/harry.jpg',
  };
  // STORE CONFIGURATION - JEST NEED IT
  const store = configureStore({
    reducer: {
      favlist: favlistReducer,
    },
  });

  test('should match the snapshot', () => {
    // SHOULD MATCH THE SNAPSHOT SO THE RENDER MUST BE A CARD
    const { container } = render(
      <Provider store={store}>
        <Card data={fakeData} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test('should show character image', () => {
    // RENDER
    render(
      <Provider store={store}>
        <Card data={fakeData} />
      </Provider>
    );
    /**
     * TO TEST IF THE CARD IS RENDERED
     * WE CAN SEARCH FOR THE MUST NOTICEABLE THING ON IT
     * THE IMAGE OF THE CHARACTER
     *
     * WE CAN KNOW THAT RENDERS THE CORRECT ONE BY
     * EXPECTING THAT THE URL IS THE SAME AS THE ONE
     * ON THE FAKE DATA
     */
    const images = screen.getAllByRole('img');
    expect(images[0].src).toBe(fakeData.image);
  });

  test('should show character info', () => {
    // RENDER
    render(
      <Provider store={store}>
        <Card data={fakeData} />
      </Provider>
    );

    /**
     * WE MUST BE SURE THAT THE CARD HAS THE CORRECT INFO
     * BY SEARCHING FOR IT ON THE HTML AS TEXT
     */
    expect(screen.getByText(fakeData.name)).toBeTruthy();
    expect(screen.getByText(fakeData.gender)).toBeTruthy();
    expect(screen.getByText(fakeData.dateOfBirth)).toBeTruthy();
    expect(screen.getByText(fakeData.eyeColour)).toBeTruthy();
    expect(screen.getByText(fakeData.hairColour)).toBeTruthy();
  });
});
