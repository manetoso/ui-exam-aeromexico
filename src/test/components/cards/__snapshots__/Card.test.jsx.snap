// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in Card.jsx should match the snapshot 1`] = `
<div>
  <div
    class="card-wrapper"
  >
    <div
      class="card-image-container card-gryffindor"
    >
      <img
        alt="."
        class="card-image"
        src="http://hp-api.herokuapp.com/images/harry.jpg"
      />
    </div>
    <div
      class="card-body"
    >
      <div
        class="card-header header-lg"
      >
        <h4
          class="card-header-label"
        >
          vivo / estudiante
        </h4>
        <button>
          <img
            alt="bookmark icon"
            class="card-header-icon"
            src="bookmark-outline.svg"
          />
        </button>
      </div>
      <h3
        class="card-title"
      >
        Harry Potter
      </h3>
      <div
        class="card-header header-sm"
      >
        <span>
          <h4
            class="card-header-label"
          >
            vivo
          </h4>
          <h4
            class="card-header-label"
          >
            estudiante
          </h4>
        </span>
        <button>
          <img
            alt="bookmark icon"
            class="card-header-icon"
            src="bookmark-outline.svg"
          />
        </button>
      </div>
      <ul
        class="card-details"
      >
        <li
          class="card-spec"
        >
          <strong>
            Cumplea√±os:
          </strong>
           
          <span>
            31-07-1980
          </span>
        </li>
        <li
          class="card-spec"
        >
          <strong>
            Genero:
          </strong>
           
          <span>
            male
          </span>
        </li>
        <li
          class="card-spec"
        >
          <strong>
            Color de ojos:
          </strong>
           
          <span>
            green
          </span>
        </li>
        <li
          class="card-spec"
        >
          <strong>
            Color de pelo:
          </strong>
           
          <span>
            black
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
